lib-extra/src/test/java/com/diffplug/spotless/extra/cpp/EclipseCdtFormatterStepTest.java

```diff
@@ -15,43 +15,31 @@
  */
 package com.diffplug.spotless.extra.cpp;
 
-import java.util.ArrayList;
-import java.util.Arrays;
-import java.util.List;
+import java.util.stream.Stream;
+
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.MethodSource;
 
-import com.diffplug.spotless.FormatterStep;
 import com.diffplug.spotless.Jvm;
 import com.diffplug.spotless.TestProvisioner;
-import com.diffplug.spotless.extra.EclipseBasedStepBuilder;
-import com.diffplug.spotless.extra.eclipse.EclipseCommonTests;
-
-class EclipseCdtFormatterStepTest extends EclipseCommonTests {
+import com.diffplug.spotless.extra.eclipse.EclipseResourceHarness;
 
-	@Override
-	protected String[] getSupportedVersions() {
-		List<String> version = new ArrayList<String>();
-		if (Jvm.version() >= 11) {
-			version.addAll(Arrays.asList("4.17.0", "4.18.0", "4.19.0", "4.20.0"));
-		}
-		version.addAll(Arrays.asList("4.7.3a", "4.11.0", "4.12.0", "4.13.0", "4.14.0", "4.16.0"));
-		return version.toArray(new String[version.size()]);
-	}
+class EclipseCdtFormatterStepTest extends EclipseResourceHarness {
+	private final static Jvm.Support<String> JVM_SUPPORT = Jvm.<String> support("Oldest Version").add(8, "4.11.0");
+	private final static String INPUT = "#include <a.h>;\nint main(int argc,   \nchar *argv[]) {}";
+	private final static String EXPECTED = "#include <a.h>;\nint main(int argc, char *argv[]) {\n}\n";
 
-	@Override
-	protected String getTestInput(String version) {
-		return "#include <a.h>;\nint main(int argc,   \nchar *argv[]) {}";
+	public EclipseCdtFormatterStepTest() {
+		super(EclipseCdtFormatterStep.createBuilder(TestProvisioner.mavenCentral()), INPUT, EXPECTED);
 	}
 
-	@Override
-	protected String getTestExpectation(String version) {
-		return "#include <a.h>;\nint main(int argc, char *argv[]) {\n}\n";
+	@ParameterizedTest
+	@MethodSource
+	void version(String version) throws Exception {
```

Comment:
What do you think about renaming `#version()` test methods to something a bit more descriptive, like `#formatsCodeWithSupportedVersion()`?

Reply:
Thanks for the heads-up. Could have sworn that I got a compilation error when I tried experimenting with JUnit 5 features.

Reply:
You are right, it is a little bit too short. But your's seems to long for me. I picked something in between...